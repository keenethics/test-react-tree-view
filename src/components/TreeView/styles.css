li.sector {
  list-style-type: none;
}

li.sector::before {
  content: '+';
  color: #d3d3d3;
  display: inline-block;
  width: 1em;
}

li.sector.expanded::before {
  content: '\2212';
}

li.sector.empty::before {
  content: '';
}

li.sector > span {
  cursor: pointer;
}

li.sector span.highlighted {
  background: gold;
  border-radius: 2px;
  padding: 0 2px;
}
