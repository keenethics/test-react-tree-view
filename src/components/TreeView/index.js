import React from 'react'
import './styles.css'

const TreeView = ({
  sectors,
  selectedId,
  selectSector,
}) => (
  <div className="tree-view">
    {sectors.map(({ name }) => (
      <li>
        {name}
      </li>
    ))}
  </div>
)

export default TreeView
