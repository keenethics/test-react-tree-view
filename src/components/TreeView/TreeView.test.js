import React from 'react'
import TreeView from './index'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

it('renders correctly', () => {
  const tree = renderer
    .create(<TreeView
      sectors={[{ id: 'id', name: 'name', items: [] }]}
      searchQuery=""
      expanded={['one', 'two']}
      selected={['one']}
      toggleView={jest.fn()}
      toggleSelection={jest.fn()}
    />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})


it('handle click correctly', () => {
  Enzyme.configure({ adapter: new Adapter() })
  const toggleView = jest.fn()
  const tree = Enzyme.mount(<TreeView
      sectors={[{ id: 'id', name: 'name', items: [] }]}
      searchQuery=""
      expanded={['one', 'two']}
      selected={['one']}
      toggleView={toggleView}
      toggleSelection={jest.fn()}
    />)
    tree.find('#id').simulate('click')
    expect(toggleView).toHaveBeenCalled()
})
