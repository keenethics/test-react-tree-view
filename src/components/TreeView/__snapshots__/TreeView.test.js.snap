// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="tree-view"
>
  <li
    className="sector "
  >
    <input
      checked={false}
      name="id"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      aria-checked={false}
      id="id"
      onClick={[Function]}
      role="checkbox"
    >
      name
    </span>
  </li>
</ul>
`;
